### DISCUSSION TITLE
5分钟在QQ群搭建ChatGPT机器人！ - Chris's blog

### DISCUSSION THREAD

**Discussion by giscus** [chrisyy2003/chrisyy2003.github.io#13] [评论]
# 5分钟在QQ群搭建ChatGPT机器人！ - Chris's blog

存在的问题 可以直接看这里排查是否存在问题  websocket bad handshake 请查看是否配置了v11适配器，在视频6：19秒处 发送指令没有回复 nonebot默认指令需要在前面加一个/斜杠来触发，请查看文章最后  前言 本文将带来快速简单在QQ群中接入ChatGPT的教程。 接入ChatGPT的安装步骤仅需三步，其中分别是：  安装cqhttp 安装nonebot框架 安装cha

https://blog.chrisyy.top/qq-chatGPT.html

**Comment by WuhuVoV** [comment]
ovo

**Comment by atlasfish** [comment]
[ERROR] nonebot_plugin_chatgpt | ChatGPT request failed: ValueError: Tag "<head>" does not correspond to any known ansi directive, make sure you did not misspelled it (or prepend '\' to escape it)
请问这个报错要怎么解决

**Reply by chrisyy2003** [reply]
openai的问题，插件很少一般不会有问题

**Comment by holygodly** [comment]
今天机器人不是加验证了嘛，群里的机器人现在是咋接的chatgpt？是在服务器反代再使用那个官网的pai吗？

**Reply by holygodly** [reply]
哦没事了，刚没看文章开头直奔评论区的。。因为之前在github上看见有个老哥用Mirai HTTP API写了个chatgpt的机器人，我以为群里用的是那个 https://github.com/lss233/chatgpt-mirai-qq-bot

**Reply by chrisyy2003** [reply]
用的都是一个接口，不行都不行。群里有个用的是api，还有个就是手动换cf字段的。

**Comment by yzyyz1387** [comment]
暗示
![image.png](https://s2.loli.net/2022/12/14/QsgpbNnX7DVjaPS.png)

**Reply by chrisyy2003** [reply]
感谢pr，晚上就合👍👍

**Comment by zqtxwd** [comment]
现在还有不需要桌面环境的方法吗?我看现在还依赖chrome

**Comment by ghost** [comment]
请问，出现 connectex: No connection could be made because the target machine actively refused it.问题该怎么修改呢，我找了好多解决方法都没有成功。一开始好好的然后突然就一直跳这个错误，然后就不能接受群消息，只能接受个人消息了。

**Reply by ghost** [reply]
又可以了，原来只要发的消息够快 就能收到

**Comment by Godleeo** [comment]
老哥，出现这个错误怎么办
请求 ChatGPT 服务器时出现问题，请稍后再试
错误信息: AttributeError: 'NoneType' object has no attribute 'new_page'

**Reply by chenmonster** [reply]
> :confused:***我也是这个错误啊***

```
Traceback (most recent call last):
  File "<string>", line 17, in <module>
  File "C:\Users\chenc\Desktop\nonebot\venv\lib\site-packages\nonebot\__init__.py", line 273, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\chenc\Desktop\nonebot\venv\lib\site-packages\nonebot\drivers\fastapi.py", line 187, in run
    uvicorn.run(
  File "C:\Users\chenc\Desktop\nonebot\venv\lib\site-packages\uvicorn\main.py", line 569, in run
    server.run()
  File "C:\Users\chenc\Desktop\nonebot\venv\lib\site-packages\uvicorn\server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "D:\Developer\Miniconda3\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "D:\Developer\Miniconda3\lib\asyncio\base_events.py", line 629, in run_until_complete
> File "C:\Users\chenc\Desktop\nonebot\venv\lib\site-packages\nonebot_plugin_chatgpt\__init__.py", line 63, in ai_chat
    msg = await chat_bot(**session[event]).get_chat_response(text)
  File "C:\Users\chenc\Desktop\nonebot\venv\lib\site-packages\nonebot_plugin_chatgpt\chatgpt.py", line 123, in get_chat_response
    async with self.get_page() as page:
  File "D:\Developer\Miniconda3\lib\contextlib.py", line 181, in __aenter__
    return await self.gen.__anext__()
  File "C:\Users\chenc\Desktop\nonebot\venv\lib\site-packages\nonebot_plugin_chatgpt\chatgpt.py", line 115, in get_page
    page = await self.content.new_page()
AttributeError: 'NoneType' object has no attribute 'new_page'
```

**Reply by chenmonster** [reply]
找到问题解决方案了，需要执行下面的命令
```sh
playwright install # 安装依赖
```

**Comment by Hrccccc** [comment]
Character AI可以植入qq群吗？求教程感谢！

**Comment by a7a8a9** [comment]
你好，我运行nb run就会有这样的报错01-16 19:55:30 [ERROR] uvicorn | Application startup failed. Exiting.请问怎么解决

**Comment by Shattered217** [comment]
总是提示token失效啊，我换了好几遍，发现每次token都会改变，所以现在是方法已经失效了吗OwO

**Reply by Shattered217** [reply]
03-18 02:31:02 [DEBUG] nonebot_plugin_chatgpt | 正在获取cf cookies
03-18 02:31:23 [ERROR] nonebot_plugin_chatgpt | cf cookies获取失败
03-18 02:31:23 [DEBUG] nonebot_plugin_chatgpt | cf cookies获取成功
03-18 02:31:23 [DEBUG] nonebot_plugin_chatgpt | 正在发送请求
03-18 02:31:23 [DEBUG] nonebot_plugin_chatgpt | 检测到session过期
03-18 02:31:23 [DEBUG] nonebot | OneBot V11 | Calling API send_msg
03-18 02:31:23 [INFO] nonebot | Matcher(type='message', module=nonebot_plugin_chatgpt.utils) running complete
03-18 02:31:23 [DEBUG] nonebot | Stop event propagation
03-18 02:31:23 [DEBUG] nonebot | Checking for matchers completed

**Comment by CCold-H** [comment]
printenv BOT_ENV得到的结果是dev，但是nb run的时候还是显示Env为prod怎么解决呢
